platform/ios/src/MGLCalloutView.h

```diff
@@ -36,6 +36,18 @@ NS_ASSUME_NONNULL_BEGIN
 @property (nonatomic, weak) id<MGLCalloutViewDelegate> delegate;
 
 /**
+ A Boolean value indicating whether the callout view should be positioned automatically.
+ The exact position can be adjusted by overriding -[UIView setCenter:]
+ */
+@property (nonatomic, assign) BOOL automaticPositioning;
```

Comment:
Positioning automatically can mean many things. I think the fundamental difference between MGLCompactCalloutView and the custom callout view that ran into this bug is that MGLCompactCalloutView is relatively positioned, while that custom callout view is absolutely positioned – kind of like `MGLRotationAlignmentMap` versus `MGLRotationAlignmentViewport`. So another possibility would be to reverse the logic and call this property `relativelyPositioned`.